c10/util/Logging.h

```diff
@@ -300,6 +300,29 @@ BINARY_COMP_HELPER(LessEquals, <=)
 C10_API void SetAPIUsageLogger(std::function<void(const std::string&)> logger);
 C10_API void LogAPIUsage(const std::string& context);
 
+// PyTorch ddp usage logging capabilities
+// DDPLoggingData holds data that can be logged in applications
+// for analysis and debugging. Data structure is defined in
+// c10 directory so that it can be easily imported by both c10
+// and torch files.
+// TODO -- right now starting with logging a small set of straightforward
+// fields, will add more fields as follow ups such as performance stats,
+// internal states and env variables and etc.
+struct DDPLoggingData {
```

Comment:
[Optional] Your design has a few logging data categories. I would suggest that we should map each data category to a separate struct inside this one. Grouping relevant fields will make this struct more manageable. Since this is the starting point, it is also fine to keep the structure flat for now.

Reply:
sounds good! let me organize the fields into separate struct in followup diff